WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -4,95 +4,78 @@ import {
 
 // let i18n-tasks know the key is used
 // i18n-tasks-use t('enums.user.role_status.delegate_regions.trainee_delegate')
 // i18n-tasks-use t('enums.user.role_status.delegate_regions.candidate_delegate')
 // i18n-tasks-use t('enums.user.role_status.delegate_regions.delegate')
 // i18n-tasks-use t('enums.user.role_status.delegate_regions.senior_delegate')
-// i18n-tasks-use t('enums.user.role_status.teams.member')
-// i18n-tasks-use t('enums.user.role_status.teams.senior_member')
-// i18n-tasks-use t('enums.user.role_status.teams.leader')
+// i18n-tasks-use t('enums.user.role_status.teams_committees.member')
+// i18n-tasks-use t('enums.user.role_status.teams_committees.senior_member')
+// i18n-tasks-use t('enums.user.role_status.teams_committees.leader')
+
+const isHyperlinkableGroup = (groupType) => groupType === GROUP_TYPE.TEAMS_COMMITTEES;
 
 export default function RolesTab({ userId, loggedInUserId }) {
-  const roleListFetch = useLoadedData(rolesOfUser(userId));
-  const loggedInUserRolesFetch = useLoadedData(rolesOfUser(loggedInUserId));
+  const roleListFetch = useLoadedData(rolesOfUser(
+    userId,
+    { isActive: true, isGroupHidden: false },
+  ));
+  const [loggedInUserRole, roleLoading, roleError] = useRoleData(
+    loggedInUserId,
+    { isActive: true },
+  );
 
   const [open, setOpen] = React.useState(false);
 
-  const activeNonHiddenRoles = React.useMemo(
-    () => {
-      if (roleListFetch.data) {
-        return roleListFetch.data.filter((role) => role.group.is_hidden === false);
-      }
-      return [];
-    },
+  const roles = React.useMemo(
+    () => (roleListFetch.data ? roleListFetch.data : []),
     [roleListFetch.data],
   );
 
-  const loggedInUserRoles = React.useMemo(
-    () => {
-      if (loggedInUserRolesFetch.data) {
-        return loggedInUserRolesFetch.data.reduce((roleMap, role) => ({
-          ...roleMap,
-          [role.group.id]: role,
-        }), {});
-      }
-      return {};
-    },
-    [loggedInUserRolesFetch.data],
-  );
-
   const isDelegate = roleListFetch.data && roleListFetch.data.some(
     (role) => role.group.group_type === 'delegate_regions',
   );
 
-  const canEditRole = (role) => role.group.group_type === 'delegate_regions' && (
-    !!loggedInUserRoles.admin || loggedInUserRoles[role.group.id]?.metadata.status === 'senior_delegate'
-  );
-
-  const canEditTeamOfRole = (role) => role.group.group_type !== 'delegate_regions' && (
-    !!loggedInUserRoles.admin || loggedInUserRoles[role.group.name]?.metadata.status === 'leader'
-  );
-
-  if (roleListFetch.loading || loggedInUserRolesFetch.loading) return <Loading />;
-  if (roleListFetch.error || loggedInUserRolesFetch.error) return <Errored />;
+  if (roleListFetch.loading || roleLoading) return <Loading />;
+  if (roleListFetch.error || roleError) return <Errored />;
 
   return (
     <>
-      {activeNonHiddenRoles.length > 0
+      {roles.length > 0
```

Comment:
```suggestion
      {roleListFetch.data?.length
```
and change the map just below to
```
            {roleListFetch.data.map((role => (
```
then you can get rid of the `roles` const definition.

Or, if you'd prefer to keep that definition, it doesn't need to be memoized and can just be
```
const roles = roleListFetch.data ?? []
```

Reply:
Oh thanks for catching that, I changed it. I was memoizing just because there was filtering earlier and I forgot to remove it after moving the filtering to server side.